# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1539914520 14400
#      Thu Oct 18 22:02:00 2018 -0400
# Node ID 92f0cc876dd6de4eb139461a87d9fc700ec97b1a
# Parent  6fec0e3052680afa728efcf8b090789ff550c390
* str-vec.cc (string_vector::delete_c_str_vec): Return early if arg is NULL.

diff --git a/liboctave/util/str-vec.cc b/liboctave/util/str-vec.cc
--- a/liboctave/util/str-vec.cc
+++ b/liboctave/util/str-vec.cc
@@ -176,16 +176,19 @@ string_vector::std_list (void) const
     retval.push_back (elem (i));
 
   return retval;
 }
 
 void
 string_vector::delete_c_str_vec (const char * const *v)
 {
+  if (! v)
+    return;
+
   const char * const *p = v;
 
   while (*p)
     delete [] *p++;
 
   delete [] v;
 }
 
